pkgs/os-specific/darwin/apple-source-releases/dtrace/0001-Patches-for-building-externally-to-Apple.patch

```diff
@@ -0,0 +1,260 @@
+@@ -1 +1 @@
+-#include "y.tab.h"
+\ No newline at end of file
++#include "dt_grammar.tab.h"
+diff --git a/lib/libdtrace/common/dt_impl.h b/lib/libdtrace/common/dt_impl.h
+index 9584acd..a8185a3 100644
+--- a/lib/libdtrace/common/dt_impl.h
++++ b/lib/libdtrace/common/dt_impl.h
+@@ -44,6 +44,7 @@
+ #endif
+ 
+ #include <sys/param.h>
++#include <sys/utsname.h>
+ 
+ #if defined(__APPLE__)
+ #define OS_LOG_SUPPORTED 1
+@@ -737,7 +738,7 @@ extern dt_pcb_t *yypcb;		/* pointer to current parser control block */
+ extern char yyintprefix;	/* int token prefix for macros (+/-) */
+ extern char yyintsuffix[4];	/* int token suffix ([uUlL]*) */
+ extern int yyintdecimal;	/* int token is decimal (1) or octal/hex (0) */
+-extern char yytext[];		/* lex input buffer */
++extern char * yytext;		/* lex input buffer */
+ extern int yylineno;		/* lex line number */
+ extern int yydebug;		/* lex debugging */
+ extern dt_node_t *yypragma;	/* lex token list for control lines */
+diff --git a/lib/libdtrace/common/dt_lex.l b/lib/libdtrace/common/dt_lex.l
+index 27f5ee1..cc3126b 100644
+--- a/lib/libdtrace/common/dt_lex.l
++++ b/lib/libdtrace/common/dt_lex.l
+@@ -699,7 +699,7 @@ if (yypcb->pcb_token != 0) {
+ void
+ yybegin(yystate_t state)
+ {
+-#ifdef	YYDEBUG
++#if	defined(YYDEBUG) && YYDEBUG
+ 	yydebug = _dtrace_debug;
+ #endif
+ 	if (yypcb->pcb_yystate == state)
+diff --git a/lib/libdtrace/common/dt_print.c b/lib/libdtrace/common/dt_print.c
+index 68e3cb2..6dc2c5f 100644
+--- a/lib/libdtrace/common/dt_print.c
++++ b/lib/libdtrace/common/dt_print.c
+@@ -77,7 +77,6 @@
+ #include <netdb.h>
+ #include <netinet/in.h>
+ #include <arpa/inet.h>
+-#include <arpa/nameser.h>
+ 
+ #include <dt_module.h>
+ #include <dt_printf.h>
+diff --git a/lib/libdtrace/common/dt_proc.c b/lib/libdtrace/common/dt_proc.c
+index 93a11c8..75f3044 100644
+--- a/lib/libdtrace/common/dt_proc.c
++++ b/lib/libdtrace/common/dt_proc.c
+@@ -92,6 +92,10 @@
+ #include <crt_externs.h>
+ #endif
+ 
++// NIX: private symbols from Darwin libproc
++int pid_suspend(int pid);
++int pid_resume(int pid);
++
+ dt_bkpt_t *
+ dt_proc_bpcreate(dt_proc_t *dpr, uintptr_t addr, dt_bkpt_f *func, void *data)
+ {
+diff --git a/lib/libdtrace/common/dt_proc.h b/lib/libdtrace/common/dt_proc.h
+index f3d62de..c15ddaa 100644
+--- a/lib/libdtrace/common/dt_proc.h
++++ b/lib/libdtrace/common/dt_proc.h
+@@ -33,6 +33,7 @@
+ #include <dt_list.h>
+ 
+ #include <sys/link.h>
++#include <rtld_db.h>
+ 
+ #ifdef	__cplusplus
+ extern "C" {
+diff --git a/lib/libproc/libproc.c b/lib/libproc/libproc.c
+index 352d49f..4b5bb63 100644
+--- a/lib/libproc/libproc.c
++++ b/lib/libproc/libproc.c
+@@ -36,7 +36,7 @@
+ #include <sys/sysctl.h>
+ 
+ #include <sys/proc_info.h>
+-#include <sys/codesign.h>
++// #include <sys/codesign.h>
+ #include <sys/fasttrap_isa.h>
+ 
+ #if DTRACE_TARGET_APPLE_MAC
+@@ -46,7 +46,7 @@
+ 
+ #include "libproc.h"
+ #include "libproc_apple.h"
+-#include "libproc_internal.h"
++// #include "libproc_internal.h"
```

Comment:
It looks like this header is present in the `xnu` source release. You could take a look at the `apple-sdk.sourceRelease` stuff the other source release packages do to get internal headers, which might shrink the patch a bit and save some declaration‐guessing. It has `sys/codesign.h`, too, and `Libc` has `os/assumes.h`.